src/main/java/tutorlink/lists/StudentList.java

```diff
@@ -0,0 +1,84 @@
+package tutorlink.lists;
+
+import tutorlink.exceptions.DuplicateMatricNumberException;
+import tutorlink.exceptions.StudentNotFoundException;
+import tutorlink.exceptions.TutorLinkException;
+import tutorlink.student.Student;
+
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class StudentList {
+
+    //Use Java String formatting to replace %s with matricNumber
+    private static final String ERROR_DUPLICATE_MATRIC_NUMBER_ON_ADD =
+            "Error! Student with Matric Number, %s, already exists in the list!";
+    private static final String TO_STRING_DELIMITER = "\n";
+    private ArrayList<Student> studentArrayList;
```

Comment:
Consider using Set instead of List as this will make it impossible to add duplicate values